frontend/src/components/SamplesList.tsx

```diff
@@ -106,22 +102,24 @@ export default function SamplesList({
     ({ userSearchVal, sampleContext }) => {
       return {
         getRows: async (params: IServerSideGetRowsParams) => {
-          let filter: DashboardRecordFilter | undefined;
+          let filters: DashboardRecordFilter[] | undefined;
           const filterModel = params.request.filterModel;
           if (filterModel && Object.keys(filterModel).length > 0) {
-            filter = {
-              field: Object.keys(filterModel)[0],
-              values: filterModel[Object.keys(filterModel)[0]].values,
-            };
+            filters = Object.entries(filterModel).map(([key, value]) => ({
+              field: key,
+              // Flexibly handle AG Grid's `any` type for filter settings by JSON.parse() this string value,
+              // then check the field name before consuming it at the GraphQL server (see https://stackoverflow.com/a/45601881)
+              filter: JSON.stringify(value),
+            }));
           } else {
-            filter = undefined; // all filter values are selected
+            filters = undefined; // all filter values are selected
           }
```

Comment:
This block is the only logic change in this file. The rest are either (1) code dedups or (2) reordering of lines to better match the new `RecordsList.tsx` to make it easier to remove duplicates between these files in the future.

Back to this highlighted block, it updates the `filter` query argument to `filters` to handle future cases where the Samples page has multiple column filters. Right now, it only has one: the "Billed" column.